src/rdkafka_topic.c

```diff
@@ -1311,7 +1332,8 @@ rd_kafka_topic_metadata_update(rd_kafka_topic_t *rkt,
         if (mdt->err == RD_KAFKA_RESP_ERR_NO_ERROR) {
                 upd += rd_kafka_topic_partition_cnt_update(rkt,
                                                            mdt->partition_cnt);
-
+                if(rd_kafka_Uuid_cmp(rkt->rkt_id, RD_KAFKA_UUID_ZERO) != 0)
```

Comment:
```suggestion
                if(rd_kafka_Uuid_cmp(rkt->rkt_id, RD_KAFKA_UUID_ZERO) == 0)
```
This needs to set it iff it's RD_KAFKA_UUID_ZERO